File: python/merge/Merger_pdf_epub.py
import aspose.pdf.HtmlSaveOptions as HtmlSaveOptions

from aspose.pdf import (
    HtmlSaveOptions,
    Document
)


def pdf_to_epub():
    path_source1 = "../../TestData/test.pdf"
    path_source2 = "../../TestData/Second/test.pdf"

    # read pdf file to Aspose Document
    first_doc = Document(path_source1)
    second_doc = Document(path_source2)

    # create empty pdf document
    output_doc = Document()

    # set less memory usage with unload instead of fast performance
    output_doc.enable_object_unload = True

    for page in first_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    for page in second_doc.pages:
        # add page from one document to another directly
        output_doc.pages.add(page)

    opt1 = HtmlSaveOptions()
    # embed css into a page
    opt1.parts_embeding_mode = HtmlSaveOptions.PartsEmbeddingModes.EMBED_ALL_INTO_HTML
    # embed images into a page
    opt1.raster_images_saving_mode = HtmlSaveOptions.RasterImagesSavingModes.AS_embedED_PARTS_OF_PNG_PAGE_BACKGROUND
    # enhance conversion of documents with backgrounds
    opt1.antialiasing_processing = HtmlSaveOptions.AntialiasingProcessingType.TRY_CORRECT_RESULT_HTML
    # use fixed layout render
    opt1.fixed_layout = True

    # save pdf to html page
    output_doc.save("test.html", opt1)

File: cpp/merge/Merger_pdf_epub.cpp
#include <iostream>

#include "Aspose.PDF.Cpp/Document.h"
#include "Aspose.PDF.Cpp/PdfFormatConversionOptions.h"
#include "Aspose.PDF.Cpp/EpubSaveOptions.h"
#include "Aspose.PDF.Cpp/EpubSaveOptions/RecognitionMode.h"
#include "Aspose.PDF.Cpp/Page.h"
#include "Aspose.PDF.Cpp/PageCollection.h"
#include "Aspose.PDF.Cpp/SaveFormat.h"

#include "system/string.h"

using namespace System;
using namespace Aspose::Pdf;

void pdf_to_epub()
{
    auto pathSource1 = u"../../TestData/test.pdf";
    auto pathSource2 = u"../../TestData/Second/test.pdf";

    // read pdf file to Aspose Document
    auto firstDoc = MakeObject<Document>(pathSource1);
    auto secondDoc = MakeObject<Document>(pathSource2);

    // create empty pdf document
    auto outputDoc = MakeObject<Document>();

    // set less memory usage with unload instead of fast performance
    outputDoc->EnableObjectUnload = true;

    for (int i = 0; i < firstDoc->get_Pages()->get_Count(); i++)
    {
        auto page = firstDoc->get_Pages()->get_Item(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->Add(page);
    }

    for (int i = 0; i < secondDoc->get_Pages()->get_Count(); i++)
    {
        auto page = secondDoc->get_Pages()->get_Item(i);
        // add page from one document to another directly
        outputDoc->get_Pages()->Add(page);
    }

    auto opt1 = MakeObject<EpubSaveOptions>();
    // keep the logical structure of transformed document
    opt1->ContentRecognitionMode = RecognitionMode::PdfFlow;
    outputDoc->Save("test.epub", opt1);
}
